### PR TITLE
feat(product): ✨ rebuild product detail page

### PR DESCRIPTION
**Pull request by sondoannam**
rebuild product detail page, handling redirect to login when click 'add to cart' button if not login... need a way to handle 'buy now' button...

## Summary by CodeRabbit

- **New Features**
	- Enhanced product detail pages with dynamic loading of detailed information, including images, variations, pricing, and technical specifications.
	- Introduced `NotFound` component for better handling of non-existent product pages.
	- Added `SelectProductVariation` component for dynamic product variation selection.
	- Implemented custom `ScrollLink` component for smooth scrolling to targets within pages.
	- New utility classes and styles for improved UI components.

- **Bug Fixes**
	- Updated cart item logic for better handling of item attributes and cart updates.
	- Fixed axios interceptor to correctly handle authorization and session management.

- **Refactor**
	- Reorganized imports and updated logic across various components for improved code clarity and performance.
	- Enhanced middleware logic for better route and authentication handling.

- **Style**
	- Introduced and adjusted utility classes and component-specific styles for alignment, justification, and button aesthetics.

- **Documentation**
	- Added comments and updated import paths to improve code readability and maintainability.

### CODE REVIEW

File: src/app/chi-tiet-san-pham/[slug]/page.tsx [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+
+    if (!product || product._id !== idExtractedFromSlug) {
+        dispatch(getDetailsProduct(idExtractedFromSlug));
+    }
+
+    if (!product) {
+        return <LoadingPage />;
+    }
+
+    return <ProductDetail product={product} />;
```
Comment (lehuygiang28) [inline src/app/chi-tiet-san-pham/[slug]/page.tsx]: @sondoannam Nếu mà k thấy sản phẩm thì loading vô cực hả

Comment (sondoannam) [reply src/app/chi-tiet-san-pham/[slug]/page.tsx]: hợp lý á

Comment (sondoannam) [reply src/app/chi-tiet-san-pham/[slug]/page.tsx]: vừa update thêm trang NotFound cho trường hợp nhập bừa url

**Review by lehuygiang28 — APPROVED** [review]
Tạm thời chưa thấy thêm vấn đề gì, good!

**Pull request merged**